wandb/sdk/artifacts/storage_policies/wandb_storage_policy.py

```diff
@@ -169,10 +172,11 @@ def load_reference(
         manifest_entry: "ArtifactManifestEntry",
         local: bool = False,
         dest_path: Optional[str] = None,
+        skip_cache: bool = False,
     ) -> Union[FilePathStr, URIStr]:
         assert manifest_entry.ref is not None
         used_handler = self._handler._get_handler(manifest_entry.ref)
-        if hasattr(used_handler, "_cache"):
+        if skip_cache and hasattr(used_handler, "_cache") and (dest_path is not None):
```

Comment:
instead should we have an `assert dest_path is not None` ?

Reply:
Calling this with `dest_path = None` is still valid, I think -- IIUC, the file is just downloaded to / loaded from the cache.  So we probably don't want that to raise an `AssertionError`

separately -- usually I'd prefer avoiding `assert` in runtime python code (i.e. outside of tests or adhoc notebooks).  If we started disallowing `dest_path = None` here, I'd rather go with one of the following patterns:
- `raise TypeError` if `dest_path is None`, or
- if possible, catch and handle (or re-raise with more info) the first error that'd be caused by having `dest_path is None`

Reply:
PS -- I think "conventional" reason I once heard for avoiding `assert` statements was that, at runtime, [they're completely ignored if Python isn't running in debug mode](https://docs.python.org/3/using/cmdline.html#cmdoption-O), e.g. running a script via `python -O my_script.py`.  TBH I've never had to do this much in practice, but 🤷

Reply:
Hm, good point. I think we can just do: ` raise TypeError if dest_path is None, or` inside the if-condition that checks if `skip_cache=True and hasattr(...)`.

I just wouldn't want the condition to fail if skip_cache=True but dest_path=None (for some highly unlikely reason), after which `load_path` would get returned with cache enabled.

Reply:
though i realize now that this is unrelated to your pr and this potential bug has existed for a while now 😅

Reply:
Nah it's definitely a good point.  Was less of an issue without the added `skip_cache` arg here -- will take a look

Reply:
Actually -- I think I'll just revert to the original signature here and get rid of the `skip_cache` arg.  It doesn't fix/improve on the the original implementation (setting `cache._override_cache_path`), just pushes it deeper and makes this part of the code more brittle :\